staging/src/k8s.io/api/autoscaling/v2/types.go

```diff
@@ -224,12 +222,12 @@ const (
 	// target (for example, transactions-processed-per-second).  The values
 	// will be averaged together before being compared to the target value.
 	PodsMetricSourceType MetricSourceType = "Pods"
-	// ResourceMetricSourceType is a resource metric known to Kubernetes, as
+	// PodResourceMetricSourceType is a resource metric known to Kubernetes, as
 	// specified in requests and limits, describing each pod in the current
 	// scale target (e.g. CPU or memory).  Such metrics are built in to
 	// Kubernetes, and have special scaling options on top of those available
 	// to normal per-pod metrics (the "pods" source).
-	ResourceMetricSourceType MetricSourceType = "Resource"
+	PodResourceMetricSourceType MetricSourceType = "Resource"
```

Comment:
suggest keeping the go name consistent with v2beta2, the field name (`resource`), the type name (`ResourceMetricSource`), and the enum value (`Resource`)

Reply:
Ok, I will try to rename it to  ResourceMetricSourceType